src/application/deployer/push_service_image.go

```diff
@@ -0,0 +1,107 @@
+package deployer
+
+import (
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/Originate/exosphere/src/aws"
+	"github.com/Originate/exosphere/src/docker/compose"
+	"github.com/Originate/exosphere/src/types"
+	"github.com/aws/aws-sdk-go/service/ecr"
+)
+
+// PushServiceImageOptions is the options to PushServiceImage
+type PushServiceImageOptions struct {
```

Comment:
How are we defining the difference between `options` and `config` in this codebase? I think we might be using both interchangeably because `config` seems to make more sense to me here. There is added confusion too because now it seems we have a `PushImageHelper` struct with very similar fields. Would it be possible to combine those structs at all?

Reply:
Options / Config isn't very different to me. For objects that are only used as function arguments (and have not alternative meaning), I think the Options suffix is a good idea. Config is more for things that are actually user configuration. That said, the comment should describe well what it represents. 

The `PushImageHelper` is ECR specific and doesn't know about our application types. It has its own API that is specific to what its doing. This is attempting to make the break between when we are using ECR and our application logic. I think it makes it easier to think about and allows us to switch out ECR if we wanted to host our images elsewhere.